ginivision/src/main/java/net/gini/android/vision/PaymentData.java

```diff
@@ -90,28 +137,34 @@ public boolean equals(final Object o) {
 
         final PaymentData that = (PaymentData) o;
 
-        if (!mAmount.equals(that.mAmount)) {
+        if (!mUnparsedContent.equals(that.mUnparsedContent)) {
+            return false;
+        }
+        if (mAmount != null ? !mAmount.equals(that.mAmount) : that.mAmount != null) {
             return false;
         }
-        if (!mBIC.equals(that.mBIC)) {
+        if (mBIC != null ? !mBIC.equals(that.mBIC) : that.mBIC != null) {
             return false;
         }
-        if (!mIBAN.equals(that.mIBAN)) {
+        if (mIBAN != null ? !mIBAN.equals(that.mIBAN) : that.mIBAN != null) {
             return false;
         }
-        if (!mPaymentRecipient.equals(that.mPaymentRecipient)) {
+        if (mPaymentRecipient != null ? !mPaymentRecipient.equals(that.mPaymentRecipient)
```

Comment:
I would extract this to a separated method, it is more clear and less prone to errors.

Reply:
equals() and hashCode() are boilerplate generated methods. I would leave them as they are.